### DISCUSSION TITLE
how to disable input sorting?

### DISCUSSION THREAD

**Discussion by GDATTACKER-RESEARCHER** [projectdiscovery/httpx#1572] [Q&A] [answered]
![image](https://github.com/projectdiscovery/httpx/assets/37478652/969af876-2866-42c7-a388-17e0d9bc4f70)
 why is it automatically sorting the input while it need to take as it is given same with dnsx tool can you check for the test i have used alexa 1 million list

**Comment by ehsandeep** [comment, marked as answer]
input sorting is done in most of PD tools for dedupping targets from input list default, if you wanted to disable sorting, you can use `-stream` mode!

```
   -s, -stream                   stream mode - start elaborating input targets without sorting
```

if you wanted to disable dedup check, you can use `-sd` option

```
   -sd, -skip-dedupe  disable dedupe input items (only used with stream mode)
```
